### PR TITLE
types: fix incorrect `weekday` for `ALLOW_INVALID_DATES` mode

### PR DESCRIPTION
**Pull request by b41sh**
### What problem does this PR solve? 

Fix the problem invalid `weekday` function
[#10835](https://github.com/pingcap/tidb/issues/10835)

### What is changed and how it works?

ignore GoTime error

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: types/mytime.go [hunk 1]
```diff
@@ -70,8 +70,9 @@ func (t MysqlTime) Microsecond() int {
 func (t MysqlTime) Weekday() gotime.Weekday {
 	// TODO: Consider time_zone variable.
 	t1, err := t.GoTime(gotime.Local)
+	// allow invalid dates
 	if err != nil {
-		return 0
+		return t1.Weekday()
```
Comment (imtbkcat) [inline types/mytime.go]:
```suggestion
		t1, _ := t.GoTime(gotime.Local)
		return t1.Weekday()
```
I think this is better.

Comment (b41sh) [reply types/mytime.go]: ok

**Comment by cyliu0** [comment]
@b41sh We may need to cherry pick this one to release branches including release-3.1.

**Pull request merged**